src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeMetadataNode.cs

```diff
@@ -48,9 +48,13 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
             {
                 // A delegate type metadata is rather useless without the Invoke method.
                 // If someone reflects on a delegate, chances are they're going to look at the signature.
```

Comment:
I must admit that I'm not a fan of this comment :wink:
It sounds like "well, let's try to make the app work" - which goes against lot of our other efforts which are more like "we will make the app work if we know... if we don't we'll warn, not magic".

I would much prefer we simply write down some specific decisions we've made (either in AOT or linker), one of which is:
* Keep Invoke method as a whole on a delegate (since it's hard to figure out if the delegate is ever called)

Similar to other rules like:
* Keep all static fields on enums - so that we can make ToString work.
* Keep all fields on Attribute - until we reimplement comparison logic for them
* Keep operators around if we see usage of Linq.Expressions - since we don't have a solution without this

There are many more things like this where we "gave up" on a problem for now.